src/Media.Plugin.iOS/MediaImplementation.cs

```diff
@@ -141,9 +155,13 @@ public Task<MediaFile> TakeVideoAsync(StoreVideoOptions options)
             if (!IsCameraAvailable)
                 throw new NotSupportedException();
 
-            CheckCameraUsageDescription();
+            CheckUsageDescription(cameraDescription, microphoneDescription);
 
-            VerifyCameraOptions(options);
+			var permissionsToCheck = new List<Permission> { Permission.Camera, Permission.Microphone };
+			if (options.SaveToAlbum)
+				permissionsToCheck.Add(Permission.Photos);
+
+			VerifyCameraOptions(options);
 
```

Comment:
`TakeVideoAsync` method is missing a call to check permission

`await CheckPermissions(permissionsToCheck.ToArray());`